lightning/src/ln/channelmonitor.rs

```diff
@@ -1668,22 +1668,23 @@ impl<ChanSigner: ChannelKeys> ChannelMonitor<ChanSigner> {
 	}
 
 	/// Attempts to claim a remote HTLC-Success/HTLC-Timeout's outputs using the revocation key
-	fn check_spend_remote_htlc(&mut self, tx: &Transaction, commitment_number: u64, height: u32) -> Vec<ClaimRequest> {
+	fn check_spend_remote_htlc(&mut self, tx: &Transaction, commitment_number: u64, height: u32) -> (Vec<ClaimRequest>, (Sha256dHash, Vec<TxOut>)) {
 		//TODO: send back new outputs to guarantee pending_claim_request consistency
+		let htlc_txid = tx.txid(); //TODO: This is gonna be a performance bottleneck for watchtowers!
```

Comment:
I think this comment is bogus - it refers to the fact that watchtowers, in some other .txid() calls loop over each monitor and for each monitor calculate each transactions' txid again and again. In this case, check_spend_remote_htlc is only called if the prevout txid matches in block_connected, so you can drop the TODO.